packages/oft-evm/contracts/OFTFee.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import { ERC20, IERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { IFee } from "./interfaces/IFee.sol";
+import { OFTCore } from "./OFTCore.sol";
+import { OFT } from "./OFT.sol";
+import { Fee } from "./Fee.sol";
+
+/**
+ * @title OFTFee Contract
+ * @dev OFT is an ERC-20 token that extends the functionality of the OFTCore contract.
+ */
+abstract contract OFTFee is OFT, Fee {
+    /**
+     * @dev Constructor for the OFT contract.
+     * @param _name The name of the OFT.
+     * @param _symbol The symbol of the OFT.
+     * @param _lzEndpoint The LayerZero endpoint address.
+     * @param _delegate The delegate capable of making OApp configurations inside of the endpoint.
+     */
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        address _lzEndpoint,
+        address _delegate
+    ) OFT(_name, _symbol, _lzEndpoint, _delegate) {}
+
+    /**
+     * @notice Indicates whether the OFT contract requires approval of the 'token()' to send.
+     * @return requiresApproval Needs approval of the underlying token implementation.
+     *
+     * @dev In the case of OFT where the contract IS the token, approval is NOT required.
+     */
+    function approvalRequired() external pure virtual override returns (bool) {
```

Comment:
Shouldnt need this, should come with the oft inheretence?

Reply:
Yes, but it is msg sender, thus they will always be being called from an approved spender, and is guarded via the slippage input

Reply:
This was done because if approval is not granted (i.e., https://github.com/LayerZero-Labs/devtools/blob/feat/oft-with-fee/packages/oft-evm/test/OFTFee.t.sol#L282) then it will revert:

```
revert: ERC20: insufficient allowance
```